src/cc/lib/distributed/graph_engine.cc

```diff
@@ -275,10 +300,15 @@ grpc::Status GraphEngineServiceImpl::GetNodeStringFeatures(::grpc::ServerContext
             continue;
         }
 
-        const auto index = internal_id->second;
-        m_partitions[m_partitions_indices[index]].GetNodeStringFeature(
-            m_internal_indices[index], features, dimensions.subspan(features_size * node_offset, features_size),
-            values);
+        auto index = internal_id->second;
+        size_t partition_count = m_counts[index];
+        bool found = false;
+        for (size_t partition = 0; partition < partition_count && !found; ++partition, ++index)
+        {
+            found = m_partitions[m_partitions_indices[index]].GetNodeStringFeature(
+                m_internal_indices[index], features, dimensions.subspan(features_size * node_offset, features_size),
```

Comment:
nit: same as above, `dimensions.subspan(features_size * node_offset, features_size)` doesn't change in the loop and can be created outside.

Reply:
same below